### PR TITLE
Automatically an integration build on CI

### LINKED ISSUE #20: Run an integration build for every PR
**Is your feature request related to a problem?**

CircleCI does not report if a Pull Request cannot be merged into master. GitHub will prevent you from merging a pull request if it has conflicts against target branch but it doesn't run tests after the merge is complete.

**Describe the solution you'd like**

We want a job to run for every commit that will automatically rebase against master and run tests. If tests fail, report that as errors.

### PR DESCRIPTION
**Pull request by taras**
Closes #20 

## Purpose

We want to make sure that it'll still pass tests after the pull request is merged into master.

## Approach

In this PR we introduce two distinct builds: "Commit Build" and "PR Build". "Commit Build" will checkout the latest commit in the branch and run tests. "PR Build" will checkout a branch where the target branch is already merged and run tests against the merged code.

To distinguish the two types of builds, this PR adds `use-merge` parameter to `build-ios` and `build-android` builds. Passing any non empty string to `use-merge` will tell the build to checkout the merged branch. 

## Learning

GitHub creates reference branches that have merged code against the target branch. You can find more about this [here](https://github.community/t5/How-to-use-Git-and-GitHub/Checkout-a-branch-from-a-fork/td-p/77).

## Screenshots

image

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -56,11 +56,25 @@ commands:
             mkdir -p ~/.local/share/.nativescript-cli
             cp user-settings.json ~/.local/share/.nativescript-cli
 
+  checkout-merged:
+    steps:
+      - run:
+          name: "Add reference to Pull Requests' merge branch as a local branch"
```
Comment (cherewaty) [inline .circleci/config.yml]: Move parentheses, not sure pull request needs to be capitalized: `Add reference to pull request's merge branch as a local branch`

**Comment by taras** [comment]
@cowboyd CircleCI doesn't provide a way to distinguish a Pull Request Build from a Branch Build. 

Ideally, I would be able to skip the integration workflow when running a Branch Build but I don't have any way to do this for a workflow. I can only check if `CIRCLE_PULL_REQUEST` is available at runtime. I changed the script to not fail the build when running a Branch Build but I can't prevent the rest of the job from running.

**Comment by cowboyd** [comment]
This is awesome! 🎉 

Is there a way we can install a custom webhook to kick off this build when the upstream head changes?

**Comment by taras** [comment]
> Is there a way we can install a custom webhook to kick off this build when the upstream head changes?

Maybe a probot?

**Pull request merged**